src/main/java/org/mifos/connector/tnm/camel/routes/PayBillRouteProcessor.java

```diff
@@ -196,46 +197,48 @@ public void processRequestForPayBillPayRoute(Exchange e) {
         e.setProperty("secondaryIdentifier", "MSISDN");
         e.setProperty("secondaryIdentifierValue", requestDto.getMsisdn());
 
-        Gson gson = new Gson();
         ChannelRequestDto channelRequestDto = TnmUtils.convertPayBillToChannelPayload(requestDto, amsName, currency);
-        String channelRequestDtoString = gson.toJson(channelRequestDto);
-        e.setProperty("PAY_REQUEST", channelRequestDtoString);
-
-        e.setProperty(CHANNEL_REQUEST, channelRequestDto);
-        e.setProperty(EXTERNAL_ID, requestDto.getTransactionId());
+        channelRequestDto.setUseWorkflowIdAsTransactionId(true);
+        channelRequestDto.setWorkflowId(oafTransactionReference);
 
         Map<String, Object> variables = new HashMap<>();
         variables.put("confirmationReceived", true);
-        variables.put(CHANNEL_REQUEST, channelRequestDtoString);
+        variables.put(CHANNEL_REQUEST, channelRequestDto.toString());
         variables.put("amount", requestDto.getTransactionAmount());
         variables.put("accountId", requestDto.getAccountNumber());
         variables.put("originDate",
                 Long.parseLong(LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMddHHmmss"))));
         variables.put("phoneNumber", requestDto.getMsisdn());
-        variables.put(SERVER_TRANSACTION_ID, requestDto.getTransactionId());
+        String tnmTransactionId = requestDto.getTransactionId();
+        variables.put(SERVER_TRANSACTION_ID, tnmTransactionId);
+        variables.put(EXTERNAL_ID, tnmTransactionId);
 
         // Getting TNM workflow transaction id and removing key
-        String transactionId = workflowInstanceStore.get(oafTransactionReference);
+        String workflowInstanceKey = workflowInstanceStore.get(oafTransactionReference);
 
-        variables.put(IS_VALIDATION_REFERENCE_PRESENT, isReconciled && transactionId != null);
-        variables.put(TRANSACTION_ID, oafTransactionReference);
+        variables.put(IS_VALIDATION_REFERENCE_PRESENT, isReconciled && workflowInstanceKey != null);
+        variables.put(TRANSACTION_ID, tnmTransactionId);
```

Comment:
About my comment on the other PR, is there any issue with setting the correct transaction ID here?